awswrangler/s3/_read_deltalake.py

```diff
@@ -0,0 +1,124 @@
+"""Amazon S3 Read Delta Lake Module (PRIVATE)."""
+
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import pandas as pd
+import pyarrow.fs as pa_fs
+from deltalake import DataCatalog, DeltaTable
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def read_deltalake(
```

Comment:
Keeping with the spirit of `awswrangler`, perhaps we should combine `read_deltalake` and `read_deltalake_from_glue` into a single method. For instance, `wr.s3.read_parquet` currently handles both reading from S3 directly or via the Glue catalog.

We did face some overloading issues with it though, so happy to be convinced otherwise, so keen to hear others' thoughts on this

Reply:
I kept the implementation separated for the moment, since `deltalake` already has its own way for getting the S3 location of a Glue table in [Rust](https://github.com/delta-io/delta-rs/blob/main/rust/src/data_catalog/glue/mod.rs#L69).